algorithms/merging/merge.py

```diff
@@ -279,3 +283,134 @@ def truncate(merged_intensities):
+        arr = anomalous_amplitudes.select(sel)
+        vals.append(dano_over_sigdano(arr))
+        resolution_bin_edges.append(anomalous_amplitudes.binner().bin_d_min(i_bin))
+    resolution_bin_edges.append(anomalous_amplitudes.binner().d_min())
+    return vals, resolution_bin_edges
+
+
+def dano_over_sigdano(anomalous_amplitudes):
+    """Calculate < |F(+) - F(-)|> / <sigma(F(+) - F(-))> i.e. DANO/SIGDANO."""
+    diff = anomalous_amplitudes.anomalous_differences()
+    if not diff.data() or not diff.sigmas():
+        return 0.0
+    return flex.mean(flex.abs(diff.data())) / flex.mean(diff.sigmas())
+
+
+def print_dano_table(anomalous_amplitudes):
+    """Calculate dano/sigdano in resolution bins and print to the logger."""
+    dFsdF, resolution_bin_edges = dano_over_sigdano_stats(anomalous_amplitudes)
+
+    logger.info("Size of anomalous differences")
+    header = ["d_max", "d_min", "dano/sigdano"]
+    rows = []
+    for i, dF in enumerate(dFsdF):
+        rows.append(
+            [
+                f"{resolution_bin_edges[i]:6.2f}",
+                f"{resolution_bin_edges[i+1]:6.2f}",
+                f"{dF:6.3f}",
+            ]
+        )
+    logger.info(tabulate(rows, header))
+
+
+def make_dano_plots(anomalous_data):
+    """
+    Make dicts of data for plotting e.g. for plotly.
+
+    Args:
+        anomalous_data (dict) : A dict of (wavelength, anomalous array) data.
+
+    Returns:
+        dict: A dictionary containing the plotting data.
+    """
+
+    data = {
+        "dF": {
+            "dano": {
+                "data": [],
+                "help": """\
+This plot shows the size of the anomalous differences of F relative to the uncertainties.
+Dano/SigDano = (< |F(+)-F(-)| > / < sigma(F(+)-F(-)) >)
+""",
+            },
+        },
+    }
+
+    for i, (wave, anom) in enumerate(anomalous_data.items()):
+        dFsdF, resolution_bin_edges = dano_over_sigdano_stats(anom)
+        d_star_sq_bins = [
+            0.5
+            * (
+                uctbx.d_as_d_star_sq(resolution_bin_edges[i])
+                + uctbx.d_as_d_star_sq(resolution_bin_edges[i + 1])
+            )
+            for i in range(0, len(resolution_bin_edges[:-1]))
+        ]
+        d_star_sq_tickvals, d_star_sq_ticktext = d_star_sq_to_d_ticks(
+            d_star_sq_bins, nticks=5
+        )
+        data["dF"]["dano"]["data"].append(
+            {
+                "x": d_star_sq_bins,
+                "y": list(dFsdF),
+                "type": "scatter",
+                "name": "\u03BB" + f"={wave:.4f}",
+            }
+        )
+
+    data["dF"]["dano"]["layout"] = {
+        "title": "Dano/SigDano vs resolution",
+        "xaxis": {
+            "title": "Resolution (Å)",
+            "tickvals": d_star_sq_tickvals,
+            "ticktext": d_star_sq_ticktext,
+        },
+        "yaxis": {"title": "Dano/SigDano", "rangemode": "tozero"},
+    }
+    return data
+
+
+def generate_html_report(mtz_file, filename):
```

Comment:
Surely it shouldn't be necessary to re-read the mtz files that we've just written in order to re-extract the anomalous arrays. Are they not already in memory somewhere? It seems a little inefficient at best.